android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.java

```diff
@@ -28,6 +30,8 @@
     public static final String LOG_TAG = RCTMGLMapViewManager.class.getSimpleName();
     public static final String REACT_CLASS = RCTMGLMapView.class.getSimpleName();
 
+    private boolean mTextureMode = false;
```

Comment:
This variable would exist across all MapViews, since the manager is only instantiated on startup, all MapViews will share this code. For the most part this is fine, but I can see a use case where we might want to have a GLSurfaceView and TextureView exist in the same app.

I think we could approach this problem slightly different. We can make a completely native view manager for TextureView support so `RCTMGLTextureMapViewManager`, and from the JS side we can have a check in the render method like

```
render () {
   if (IS_ANDROID && this.props.textureMode) {
      return <AndroidTextureMapView ... />;
   }
   return <MapView ... />;
}
```

what do you think?

Reply:
Perfect, I'm going to take a look during this week.